### DISCUSSION TITLE
changing jnp dtype

### DISCUSSION THREAD

**Discussion by backpropper** [jax-ml/jax#5784] [Q&A] [answered]
what is the recommended way to change the dtype in jnp array? I'm using something like this:
`new_jnp_arr = jnp.asarray(old_jnp_arr, dtype=jnp.int32)`

**Comment by jakevdp** [comment, marked as answer]
Using `jnp.asarray` with a `dtype` argument is fine; you can also use `new_jnp_arr = old_jnp_arr.astype(jnp.int32)`. Both approaches end up being implemented in terms of the same lower-level `jax.lax.convert_element_type` function.

**Reply by hawkinsp** [reply]
(And note: these mirror NumPy's APIs.)

**Reply by backpropper** [reply]
I see. I didn't find astype in the jax docs so though it isn't compatible. Thanks!
